liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -299,44 +288,158 @@ public ServerTask getServerTask() throws Exception {
+
             try {
-                boolean unhandledChange = false;
-                String modifiedPom = readFile(buildFile);
-                XMLUnit.setIgnoreWhitespace(true);
-                XMLUnit.setIgnoreAttributeOrder(true);
-                XMLUnit.setIgnoreComments(true);
-                DetailedDiff diff = new DetailedDiff(XMLUnit.compareXML(this.existingPom, modifiedPom));
-                List<?> allDifferences = (List<?>)diff.getAllDifferences();
-                log.debug("Number of differences in the pom: " + allDifferences.size());
-
-                for (Object differenceObj : allDifferences) {
-                    Difference difference = (Difference)differenceObj;
-                    if (difference.getControlNodeDetail().getNode() != null && !dependencyChange(difference.getControlNodeDetail().getNode())) {
-                        unhandledChange = true;
+                // Monitoring liberty properties in the pom.xml
+                Properties p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.bootstrap.");
+                Properties q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.bootstrap.");
+                if (!Objects.equals(p, q)) {
+                    restartServer = true;
+                } else {
+                    p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.jvm.");
+                    q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.jvm.");
+                    if (!Objects.equals(p, q)) {
+                        restartServer = true;
+                    } else {
+                        p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.env.");
+                        q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.env.");
+                        if (!Objects.equals(p, q)) {
+                            restartServer = true;
+                        }
                     }
-                }                
-                if (!allDifferences.isEmpty()) {
-                    log.info("Pom has been modified");
-                    if (isUsingBoost()) {
-                        log.info("Running boost:package");
-                        runBoostMojo("package", true);
+                }
+
+                if (!restartServer) {
+                    p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.var.");
+                    q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.var.");
+                    if (!Objects.equals(p, q)) {
+                        createServer = true;
+                    } else {
+                        p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.defaultVar.");
+                        q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.defaultVar.");
+                        if (!Objects.equals(p, q)) {
+                            createServer = true;
+                        }
```

Comment:
same comment as above

private boolean hasServerVariableChanged(Project project, Project backupProject)

Reply:
I can refactor it to a separate method